### PR TITLE
[bs4] More precise types for bs4.Tag.select(_one)

### PR DESCRIPTION
**Pull request by Gobot1234**
These methods annoy pyright without the annotations

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
We prefer to leave parameters for which we don't have precise types yet unannotated. What is the pyright error you're seeing?

**Comment by Gobot1234** [comment]
Oh I didn't realise they were unannotated but, I'm seeing 
```
Type of "select" is partially unknown
  Type of "select" is "(selector: str, namespaces: Any | None = ..., limit: int | None = ..., **kwargs: Unknown) -> ResultSet[Tag]"Pylance[reportUnknownMemberType]
```
I'm happy to update this PR with the correct types which I think are
```py
def select(self, selector: str, namespaces: Any | None = ..., limit: int | None = ..., flags: int = 0, *, custom: dict[str, str] | None = ...) -> ResultSet[Tag]: ...
def select_one(self, selector: str, namespaces: Any | None = ..., flags: int = 0, *, custom: dict[str, str] | None = ...) -> ResultSet[Tag]: ...
```

**Comment by JelleZijlstra** [comment]
Precise types would be even better!

File: stubs/beautifulsoup4/bs4/element.pyi [hunk 1]
```diff
@@ -329,8 +329,8 @@ class Tag(PageElement):
     def children(self) -> Iterable[PageElement]: ...
     @property
     def descendants(self) -> Iterable[PageElement]: ...
-    def select_one(self, selector: str, namespaces: Any | None = ..., **kwargs: Any) -> Tag | None: ...
-    def select(self, selector: str, namespaces: Any | None = ..., limit: int | None = ..., **kwargs: Any) -> ResultSet[Tag]: ...
+    def select_one(self, selector: str, namespaces: Any | None = ..., flags: int = 0, *, custom: dict[str, str] | None = ...) -> Tag | None: ...
+    def select(self, selector: str, namespaces: Any | None = ..., limit: int | None = ..., flags: int = 0, *, custom: dict[str, str] | None = ...) -> ResultSet[Tag]: ...
```
Comment (JelleZijlstra) [inline stubs/beautifulsoup4/bs4/element.pyi]:
```suggestion
    def select(self, selector: str, namespaces: Any | None = ..., limit: int | None = ..., *, flags: int = ..., *, custom: dict[str, str] | None = ...) -> ResultSet[Tag]: ...
```

`flags` is also kwonly, and defaults should always be `...`.

File: stubs/beautifulsoup4/bs4/element.pyi [hunk 2]
```diff
@@ -329,8 +329,8 @@ class Tag(PageElement):
     def children(self) -> Iterable[PageElement]: ...
     @property
     def descendants(self) -> Iterable[PageElement]: ...
-    def select_one(self, selector: str, namespaces: Any | None = ..., **kwargs: Any) -> Tag | None: ...
-    def select(self, selector: str, namespaces: Any | None = ..., limit: int | None = ..., **kwargs: Any) -> ResultSet[Tag]: ...
+    def select_one(self, selector: str, namespaces: Any | None = ..., flags: int = 0, *, custom: dict[str, str] | None = ...) -> Tag | None: ...
```
Comment (JelleZijlstra) [inline stubs/beautifulsoup4/bs4/element.pyi]:
```suggestion
    def select_one(self, selector: str, namespaces: Any | None = ..., *, flags: int = ..., custom: dict[str, str] | None = ...) -> Tag | None: ...
```

**Review by JelleZijlstra — COMMENTED** [review]
In theory it accepts additional kwargs: they get passed to `soupsieve.compile`, which takes `**kwargs` and ignores them. It seems more useful for type checkers to warn about additional kwargs.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by Gobot1234** [comment]
> In theory it accepts additional kwargs: they get passed to `soupsieve.compile`, which takes `**kwargs` and ignores them. It seems more useful for type checkers to warn about additional kwargs, so I think it's fine to specify only the kwargs that actually do something.

Yeah, I wasn't entirely sure what to do about that but I agree and since it doesn't affect the runtime I don't think it matters anyway.

**Pull request merged**